### PR TITLE
PP-941: print queue name in queuejob hook debug input file

### PR DESCRIPTION
**Pull request by minghui-liu**
#### Issue-ID

* **[PP-941](https://pbspro.atlassian.net/browse/PP-941)**

#### Problem description
The user request queue is not written to the queuejob hook debug input files.

#### Cause / Analysis
When queue job hooks are run, the job has not been placed into a queue yet.

#### Solution description
Appended the name of the user requested queue to the hook's debug input file.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/server/hook_func.c [hunk 1]
```diff
@@ -4169,6 +4169,17 @@ int server_process_hooks(int rq_type, char *rq_user, char *rq_host, hook *phook,
...
+		/* print name of user requested queue if I am a queuejob hook */
+		if (rq_type == PBS_BATCH_QueueJob) {
+			char *qname = ((struct rq_queuejob *)req_ptr->rq_job)->rq_destin;
+			/* use default queue if user did not specify a queue for the job */
+			if ((!qname || *qname == '\0' || *qname == '@') &&
+				server.sv_attr[SRV_ATR_dflt_que].at_flags & ATR_VFLAG_SET)
+				fprintf(fp_debug, "%s.%s=%s\n", EVENT_JOB_OBJECT, "queue",
+						server.sv_attr[SRV_ATR_dflt_que].at_val.at_str);
+			else
+				fprintf(fp_debug, "%s.%s=%s\n", EVENT_JOB_OBJECT, "queue", qname);
```
Comment (bhroam) [inline src/server/hook_func.c]: Don't put "queue" as a %s format option.  Put it directly into the format string.

File: test/tests/functional/pbs_hook_debug_input.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            self.server_hooks_tmp_dir = \
+                os.path.join(self.server.pbs_conf['PBS_HOME'],
+                             'server_priv', 'hooks', 'tmp')
+
+    def remove_files_match(self, pattern):
+        """
+        Remove hook debug files in hooks/tmp folder that
+        match pattern
+        """
+        for file in os.listdir(self.server_hooks_tmp_dir):
```
Comment (bhroam) [inline test/tests/functional/pbs_hook_debug_input.py]: file is a python builtin.  Consider using a different variable.

File: test/tests/functional/pbs_hook_debug_input.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        queuejob hook debug input file
+        """
+        hook_name = "queuejob_debug"
+        hook_body = ("import pbs\n"
+                     "pbs.event().accept()")
+        attr = {'enabled': 'true', 'event': 'queuejob', 'debug': 'true'}
+        self.server.create_import_hook(hook_name, attr, hook_body)
+
+        new_queue = 'happyq'
+        attr = {ATTR_qtype: 'e', ATTR_enable: 'True'}
```
Comment (bhroam) [inline test/tests/functional/pbs_hook_debug_input.py]: Spell out execution

File: src/server/hook_func.c [hunk 4]
```diff
@@ -4178,7 +4178,7 @@ int server_process_hooks(int rq_type, char *rq_user, char *rq_host, hook *phook,
 				fprintf(fp_debug, "%s.%s=%s\n", EVENT_JOB_OBJECT, "queue",
```
Comment (bhroam) [inline src/server/hook_func.c]: Move queue to the format string here too.

**Comment by bhroam** [comment]
@bayucan please sign off the final code.  After he does, squash and rebase for the final merge.

**Comment by minghui-liu** [comment]
Squashed and rebased.

**Comment by bhroam** [comment]
I merged in the pull request.

**Comment by minghui-liu** [comment]
Thank you @bhroam .

**Pull request merged**